plugin/src/main/java/org/jenkinsci/plugins/structs/SymbolLookup.java

```diff
@@ -133,13 +177,12 @@ public boolean equals(Object o) {
             if (this == o) return true;
             if (o == null || getClass() != o.getClass()) return false;
             Key key = (Key) o;
-            return type==key.type && tag.equals(key.tag) && name.equals(key.name);
-
+            return type.getName().equals(key.type.getName()) && tag.equals(key.tag) && name.equals(key.name);
```

Comment:
Why this change? It will mean that similarly-named classes from different class loaders will compare as equal in `Key`, which seems wrong.

Reply:
@jglick No, that's exactly the intent -- if the full classname matches, I want to ensure we don't get an incorrect cache miss due to a nested classloader or similar.

Reply:
That makes no sense that I can see. If a `Key.type` has been successfully loaded, it has a particular defining loader. Any initiating loader finding it will produce the same `Class` object. You would only have distinct classes in case the type was loaded from distinct places, such as a fork of a library, in which case the class definition could be quite different.

Was this diff line prompted by some actual experience you can point to?

Reply:
@jglick  I have in a few rare cases been bitten by classloading quirks like this, yes.  I do NOT agree that it introduces a defect, but I also don't care to fight this one out.

Reply:
I just studied the sources and I believe it would introduce a `ClassCastException` under some conditions. Unlikely ones, but the `equals` code as it was originally was both correct and faster than the variant you proposed.